File: game_server/ServerProtocol/src/GS2GC/p004_PlayerOp/GS2GC_004_024_RetRefreshChatBoxStatus.java
package GS2GC.p004_PlayerOp;

import java.nio.ByteBuffer;
public class GS2GC_004_024_RetRefreshChatBoxStatus implements ALBasicProtocolPack._IALProtocolStructure {
/** 宝箱实例ID */
private long instanceId;
/** 宝箱状态 */
private NPEnum.ENPBoxChatStatus boxStatus;
/** 已经领取的玩家CID列表 */
private java.util.ArrayList<Long> gainedCidList;


public GS2GC_004_024_RetRefreshChatBoxStatus() {
	instanceId = (long)0;
	boxStatus = NPEnum.ENPBoxChatStatus.values()[0];
	gainedCidList = new java.util.ArrayList<Long>();
}

public GS2GC_004_024_RetRefreshChatBoxStatus(
	 long _instanceId
	, NPEnum.ENPBoxChatStatus _boxStatus
	, java.util.ArrayList<Long> _gainedCidList
) {	instanceId = _instanceId;
	boxStatus = _boxStatus;
	gainedCidList = _gainedCidList;
}

public final byte getMainOrder() { return (byte)4; }

public final byte getSubOrder() { return (byte)24; }

/** 宝箱实例ID */
public long getInstanceId() { return instanceId; }
/** 宝箱实例ID */
public void setInstanceId(long _instanceId) { instanceId = _instanceId; }
/** 宝箱状态 */
public NPEnum.ENPBoxChatStatus getBoxStatus() { return boxStatus; }
/** 宝箱状态 */
public void setBoxStatus(NPEnum.ENPBoxChatStatus _boxStatus) { boxStatus = _boxStatus; }
/** 已经领取的玩家CID列表 */
public java.util.ArrayList<Long> getGainedCidList() { return gainedCidList; }
/** 已经领取的玩家CID列表 */
public void addGainedCidList(long _gainedCidList) { gainedCidList.add(_gainedCidList); }


public final int GetBufSize() {
	int _size = 12;
	_size += 2 + (gainedCidList.size() * 8);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 14;
	_size += 2 + (gainedCidList.size() * 8);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) instanceId = _buf.getLong();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) boxStatus = NPEnum.ENPBoxChatStatus.ENPBoxChatStatus_FromInt(_buf.getInt());
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _gainedCidListCount = _buf.getShort();
	for(int _i = 0; _i < _gainedCidListCount; _i++) { 
		long _gainedCidList = (long)0;
		if(_buf.remaining() > 0) _gainedCidList = _buf.getLong();
		gainedCidList.add(_gainedCidList);
	}
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putLong(instanceId);
	_buf.putInt(boxStatus.ordinal());

	_buf.putShort((short)gainedCidList.size());
	for(int _i = 0; _i < gainedCidList.size(); _i++) { 
		_buf.putLong(gainedCidList.get(_i));
	}
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)24);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)24);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

File: game_client/Assets/Scripts/ClientProtocol/GS2GC/p004_PlayerOp/GS2GC_004_024_RetRefreshChatBoxStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p004_PlayerOp
{

public class GS2GC_004_024_RetRefreshChatBoxStatus : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 宝箱实例ID
/// </summary>
private long instanceId;
/// <summary>
/// 宝箱状态
/// </summary>
private NPEnum.ENPBoxChatStatus boxStatus;
/// <summary>
/// 已经领取的玩家CID列表
/// </summary>
private List<long> gainedCidList;


public GS2GC_004_024_RetRefreshChatBoxStatus() {
	instanceId = (long)0;
	boxStatus = 0;
	gainedCidList = new List<long>();
}

public GS2GC_004_024_RetRefreshChatBoxStatus(
	long _instanceId
	, NPEnum.ENPBoxChatStatus _boxStatus
	, List<long> _gainedCidList
) {	instanceId = _instanceId;
	boxStatus = _boxStatus;
	gainedCidList = _gainedCidList;
}

public byte getMainOrder() { return (byte)4; }

public byte getSubOrder() { return (byte)24; }

/// <summary>
/// 宝箱实例ID
/// </summary>
public long getInstanceId() { return instanceId; }
/// <summary>
/// 宝箱实例ID
/// </summary>
public void setInstanceId(long _instanceId) { instanceId = _instanceId; }
/// <summary>
/// 宝箱状态
/// </summary>
public NPEnum.ENPBoxChatStatus getBoxStatus() { return boxStatus; }
/// <summary>
/// 宝箱状态
/// </summary>
public void setBoxStatus(NPEnum.ENPBoxChatStatus _boxStatus) { boxStatus = _boxStatus; }
/// <summary>
/// 已经领取的玩家CID列表
/// </summary>
public List<long> getGainedCidList() { return gainedCidList; }
/// <summary>
/// 已经领取的玩家CID列表
/// </summary>
public void addGainedCidList(long _gainedCidList) { gainedCidList.Add(_gainedCidList); }


public int GetBufSize() {
	int _size = 12;
	_size += 2 + (gainedCidList.Count * 8);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 14;
	_size += 2 + (gainedCidList.Count * 8);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	instanceId = _buf.getLong();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	boxStatus = (NPEnum.ENPBoxChatStatus)_buf.getInt();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _gainedCidListCount = _buf.getShort();
	for(int _i = 0; _i < _gainedCidListCount; _i++) { 
		long _gainedCidList = (long)0;
		_gainedCidList = _buf.getLong();
		gainedCidList.Add(_gainedCidList);
	}
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putLong(instanceId);
	_buf.putInt((int)boxStatus);

	_buf.putShort((short)gainedCidList.Count);
	for(int _i = 0; _i < gainedCidList.Count; _i++) { 
		_buf.putLong(gainedCidList[_i]);
	}
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)24);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)24);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("instanceId").Append(":").Append(instanceId.ToString()).Append(", ");
	builder.Append("boxStatus").Append(":").Append(boxStatus.ToString()).Append(", ");
	builder.Append("gainedCidList").Append(":").Append(gainedCidList.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

